### DISCUSSION TITLE
Tool to check and increase the percentage of zero value pixels in a wallpaper.

### DISCUSSION THREAD

**Discussion by AyanBhunia** [Hash-Studios/Prism#246] [Ideas]
Personally, I prefer wallpapers that can use the AMOLED screen and save some battery life. Therefore, every time I change my wallpaper, I have to use another tool to check if the black pixels are actually #000000. 

You can display a percentage to show how many pixels in a particular wallpaper have a zero value. Additionally, it would be helpful if you added a tool that changes nearest points to zero; and we could adjust the depth of the change.

Like this AMOLED wallpaper downloaded from internet [source](https://www.google.com/url?sa=i&url=http%3A%2F%2Fm.gettywallpapers.com%2Famoled-hd-wallpapers%2F&psig=AOvVaw3wy459WqofcrbBaxnEGFkq&ust=1638470021664000&source=images&cd=vfe&ved=0CAwQjhxqFwoTCIDByM-ew_QCFQAAAAAdAAAAABAD) :

![a](https://user-images.githubusercontent.com/46602405/144292862-3d61068b-c2aa-4c14-80a7-cf6000937061.jpg)

This have only 768 zero value pixels out of 2592000. This means that the entire red color area of the image will be displayed.
Turn off percentage: 0.03%

![Collage](https://user-images.githubusercontent.com/46602405/144310034-96915dfa-4e49-456d-825c-c4d289b04fcb.jpg)

In the application you can show the percentage of zero value pixels in images. Also , a tool to change near zero values to zero with a slider .
The tool can ;**method 1** just modify every pixel to zero if value is smaller than x. For bigger x or depth we will lost the in between darker pixels.

for x = 5. 
Turn off percentage: 23.5%.  

![Collage](https://user-images.githubusercontent.com/46602405/144391227-59bb7dd8-dcc0-4ef1-8d0f-bea89ddb4069.jpg)

left is original image ,the black area from middle image will turn off screen pixels and right image is modified.

for x = 15. 
Turn off percentage: 69.6%. 

![Collage](https://user-images.githubusercontent.com/46602405/144392246-bc385253-e4d8-4104-b11a-1a6f14012528.jpg)

**Method 2** , the tool can use a weighted function which will modify pixels to zero if smaller than x and decrease other pixels value more if they are near x. Using simple function like pixel = (pixel - x*x/pixel) will give.

For x = 5, 
Turn off percentage: 23.5%.  

![Collage](https://user-images.githubusercontent.com/46602405/144393253-b8efbe45-1197-4ff9-bd21-a8648887a635.jpg)

For x = 15, 
Turn off percentage: 69.6%.  

![Collage](https://user-images.githubusercontent.com/46602405/144393417-a46638f9-0e5e-4637-b868-92a2bd7516a2.jpg)

For x = 25, 
Turn off percentage: 91.3%.  

![Collage](https://user-images.githubusercontent.com/46602405/144393674-7e0d96f2-c57c-4f49-bfcc-22fc16f0c840.jpg)

If I were good at mobile app development, I would do it myself. So, it will be great if you add this feature in the app.

**Comment by codenameakshay** [comment]
Woah, that's quite insightful. I will try to implement this in the next update, this is a great idea.
